### PR TITLE
Problem analysis

### PR DESCRIPTION
**Pull request by TheRumle**
**Problem analysis**
The problem analysis now focus on the initial interview. Specifically, it focuses on how we conduct the interviews and why we select the interview types. This leads into a section describing the stakeholders current problems - here focus is on three key problems: 

1. Not enough available rooms for drop in meetings due to unused meeting rooms. 
2. Registering new meetings is tedious since they have to do it from their pc.
3. Making additional arrangement for meetings is not integrated into their current workflow - many different systems are used. 

What needs to be done now, is to figure out and write about the concrete direction we want to go in. This include examining the problem further (based on the developed prototype), and figure out which of these problems we want to handle and why that problem is the one we choose to solve. This should lead into a problem goal, and the technology researched in state of the art could be used to define a problem statement. For instance, if we find out we can use technology x to figure out when a room is unused, we can define a problem statement based on x.
The chapter needs to be expanded after the requirements section - this expansion should examine state of the art and potential technologies to use in the solution.

**Read-through**
The report looks really nice now. There was some sentences that I re-wrote, often due to using spoken-language instead of technical description. My general thoughts are:

- We need to be aware of the language we use. Often things are convoluted and spoken language.
- When restructuring the report ensure that things are cited. I found subsubsections without a single cite. Better to cite twice than once when in doubt. 
- Using * when creating subsections makes it a bit hard to refer back and forth to sections. We should consider not using them. I'd rather have a long list of contents and accurate references than reference to the beginning of a 3 page section.

### CODE REVIEW

File: report/src/introduction/agile_approach.tex [hunk 1]
```diff
@@ -1,29 +1,27 @@
 \section{Implementing Agile Project Management with Scrum}
 Developing a software product for a business can be a complex process.
-Incremental agile approaches have become popular to address this complexity, as they allow for faster development and adaptation to changes.\cite{sommervilleSoftwareEngineering2016}
-In incremental agile approaches, a system is developed as a series of increments, each adding functionality to the system.
-Between these increments, one gathers feedback from users to prioritize functionality for the next increment.
+Incremental agile approaches have become popular to address this complexity, as it allows for fast adaptation to changes in user needs. \cite{sommervilleSoftwareEngineering2016}
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:3]: Writing `it` is wrong as the sentence refers to multiple agile approaches.
Narrowing down to user needs is also, in my view, wrong. You learn about the problem as you explore solutions. That doesn't just include users changing needs.

File: report/src/introduction/agile_approach.tex [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
+In incremental-agile approaches, system development is considered a series of increments, each adding functionality to the system.
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:4]: Why add the dash now and not elsewhere?
Don't think this change is better. The previous describes the approach better imo.

File: report/src/introduction/agile_approach.tex [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
+Between these increments, one gathers feedback from stakeholders to prioritize functionality for the next increment.
 Therefore, each increment adds value to the user.
-By continuously getting feedback and working with these increments, the development process becomes cheaper and faster as a result of less time spent reworking the system.\cite{sommervilleSoftwareEngineering2016}
-This section will discuss how we are using an agile approach for our project, focusing specifically on the Scrum framework.
+By continuously getting feedback and working with these increments, the development process becomes cheaper and faster as less time is spent adapting to changes.\cite{sommervilleSoftwareEngineering2016}
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:7]: Ambivalent on this one. It's still correct, and I like the brevity. However, this _seems_ to indicate that adapting to changes is what we wish to avoid... and that's not correct.

File: report/src/introduction/agile_approach.tex [hunk 4]
```diff
@@ -1,29 +1,27 @@
...
-\subsection*{Scrum Framework}
-Agile is a set of values, and does not present practical activities which a process model would consist of\cite{sutherlandScrumArtDoing2014}.
-Therefore, the team has decided on Scrum\cite{scrumdotorg}, which is an agile framework that puts the values to practice.
-This decision was made due to prior experience with using the framework amongst team members.
+\subsection{Scrum Framework}
+Agile is a set of values and does not present practical activities which a process model would consist of{sutherlandScrumArtDoing2014}.
+Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}
 
-Scrum introduces a variety of new practices and roles, which we will describe in the following sections.
+Scrum introduces a number of practices and roles, which is described in the following sections.
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:14]: `is` -> `are`?
Is the change in tense intentional? Not used elsewhere.

File: report/src/introduction/agile_approach.tex [hunk 5]
```diff
@@ -1,29 +1,27 @@
...
+\subsection{Scrum Framework}
+Agile is a set of values and does not present practical activities which a process model would consist of{sutherlandScrumArtDoing2014}.
+Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}
 
-Scrum introduces a variety of new practices and roles, which we will describe in the following sections.
+Scrum introduces a number of practices and roles, which is described in the following sections. 
 
 \subsubsection*{Roles and Responsibilities}
-In a Scrum team, there is a Scrum Master, a Product Owner, and developers.
+A Scrum team consists of three roles: the Scrum Master, a Product Owner, and developers.
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:17]: Not sure leading with a colon is great. In general, using it to list things in this kind of report makes the writing seem lazier. <- IMO

File: report/src/introduction/agile_approach.tex [hunk 6]
```diff
@@ -1,29 +1,27 @@
...
+By continuously getting feedback and working with these increments, the development process becomes cheaper and faster as less time is spent adapting to changes.\cite{sommervilleSoftwareEngineering2016}
+This section will discuss how an agile approach has been utilized in this project. 
 
-\subsection*{Scrum Framework}
-Agile is a set of values, and does not present practical activities which a process model would consist of\cite{sutherlandScrumArtDoing2014}.
-Therefore, the team has decided on Scrum\cite{scrumdotorg}, which is an agile framework that puts the values to practice.
-This decision was made due to prior experience with using the framework amongst team members.
+\subsection{Scrum Framework}
+Agile is a set of values and does not present practical activities which a process model would consist of{sutherlandScrumArtDoing2014}.
+Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:12]: `Scrum chosen` - looks like a word is missing.

File: report/src/introduction/agile_approach.tex [hunk 7]
```diff
@@ -33,18 +31,22 @@ \subsubsection*{Scrum Artifacts and Tools}
...
+User stories represent functionality the stakeholder would like to see in the system.
 For this reason, they must include \textit{who} it is for, \textit{what} should be done, and \textit{why} they want it.
 This is captured in the general "\texttt{As a X, I want Y, such that Z.}" format for user stories.
 A \textbf{sprint} is a short period of time, during which the team works on a subset of the product backlog, captured in a \textbf{sprint backlog}.
 This defines the \textit{what} of the sprint.
 The items in the sprint backlog are chosen based on a \textbf{sprint goal}, defining the \textit{why} of the sprint.
-The sprint also has an actionable plan for delivering the increment, representing the \textit{how}.
+
+
+The sprint has an actionable plan for delivering the increment, representing the \textit{how}. \todo{what is how? It is not mentioned anywhere else.}
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:43]: `What`, `how`, and `why` are referenced throughout the three previous sentences to denote the functions that the aspects presented have. To say that a sprint backlog is the `what` means that it denotes what should be done. Similarly, saying that the sprint goal denotes the `why` is to say that the spring goal denotes why the sprint is being undertaken.
And finally, `how` would then be represented by the actionable plan to deliver the increment.
That is to say, we know `what` we are doing, we know `why` we are doing it, and we know `how` to do it.

File: report/src/introduction/agile_approach.tex [hunk 8]
```diff
@@ -1,29 +1,27 @@
...
-By continuously getting feedback and working with these increments, the development process becomes cheaper and faster as a result of less time spent reworking the system.\cite{sommervilleSoftwareEngineering2016}
-This section will discuss how we are using an agile approach for our project, focusing specifically on the Scrum framework.
+By continuously getting feedback and working with these increments, the development process becomes cheaper and faster as less time is spent adapting to changes.\cite{sommervilleSoftwareEngineering2016}
+This section will discuss how an agile approach has been utilized in this project. 
 
-\subsection*{Scrum Framework}
-Agile is a set of values, and does not present practical activities which a process model would consist of\cite{sutherlandScrumArtDoing2014}.
-Therefore, the team has decided on Scrum\cite{scrumdotorg}, which is an agile framework that puts the values to practice.
-This decision was made due to prior experience with using the framework amongst team members.
+\subsection{Scrum Framework}
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:10]:
```suggestion
\subsection*{Scrum Framework}
```

Comment (chhoumann) [reply report/src/introduction/agile_approach.tex:10]: Or the ToC looks like this:
image

File: report/src/introduction/agile_approach.tex [hunk 9]
```diff
@@ -53,17 +55,15 @@ \subsubsection*{Scrum Artifacts and Tools}
 Daily scrums are meetings where progress towards the sprint goal is discussed.
 Each increment will be inspected during \textbf{sprint review}.
 During such a review, progress is examined, the team and its stakeholders discuss any changes, and then figure out what to do next.
+
 Finally, the team conducts a \textbf{sprint retrospective}, where the team examines how their last sprint went, and how they may improve.
 This is part of the quest for continuous improvement.\cite{sutherlandScrumArtDoing2014}
 
-\subsubsection*{Scrum Process}
-In short, the process for working as a Scrum team involves repeatedly taking backlog items and working on them during a sprint.
+In summary, Scrum involves repeatedly taking backlog items and working on them during a sprint.
```
Comment (chhoumann) [inline report/src/introduction/agile_approach.tex:62]: I think it makes sense to remove the section, but I really like the aesthetics of having it. It also, to me, improves the reading process by separating the summary & DoDs into their own section. 
`Artifacts and Tools` becomes very long, and its scope gets a little large.

Thoughts?

File: report/src/introduction/premise.tex
Comment (chhoumann) [inline report/src/introduction/premise.tex:1]: Make sure to finalize this before merging. Very apparent that it's a draft :)

File: report/src/problem_analysis/user_stories.tex
Comment (chhoumann) [inline report/src/problem_analysis/user_stories.tex:1]: File name doesn't seem to match its contents anymore

**Review by chhoumann — CHANGES_REQUESTED** [review]
Great PR, guys!
Overall, I can understand where you're going with this. I think this really improves the report.

I neglected reviewing `user_stories.tex` and `premise.tex` (see my comments).

File: report/src/introduction/agile_approach.tex (hunk 7 above)
Comment (TheRumle) [inline report/src/introduction/agile_approach.tex:43]:
```suggestion
To establish how the user story gets completed, an actionable plan for delivering the increment is made. This plan is produced during the start of each sprint in a process called \textbf{sprint planning}.
```

File: report/src/introduction/agile_approach.tex [hunk 10]
```diff
@@ -33,18 +31,22 @@ \subsubsection*{Scrum Artifacts and Tools}
...
 For this reason, they must include \textit{who} it is for, \textit{what} should be done, and \textit{why} they want it.
 This is captured in the general "\texttt{As a X, I want Y, such that Z.}" format for user stories.
 A \textbf{sprint} is a short period of time, during which the team works on a subset of the product backlog, captured in a \textbf{sprint backlog}.
 This defines the \textit{what} of the sprint.
 The items in the sprint backlog are chosen based on a \textbf{sprint goal}, defining the \textit{why} of the sprint.
-The sprint also has an actionable plan for delivering the increment, representing the \textit{how}.
+
+
+The sprint has an actionable plan for delivering the increment, representing the \textit{how}. \todo{what is how? It is not mentioned anywhere else.}
 At the beginning of teach sprint, the Scrum team performs \textbf{sprint planning}.
```
Comment (TheRumle) [inline report/src/introduction/agile_approach.tex:44]: If other suggestion is okay this needs to be removed
```suggestion
```

File: report/src/introduction/agile_approach.tex (hunk 6 above)
Comment (TheRumle) [inline report/src/introduction/agile_approach.tex:12]:
```suggestion
Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum is chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}
```

File: report/src/introduction/agile_approach.tex (hunk 2 above)
Comment (TheRumle) [inline report/src/introduction/agile_approach.tex:4]:
```suggestion
In incremental agile approaches, system development is considered a series of increments, each adding functionality to the system.
```

Comment (TheRumle) [reply report/src/introduction/agile_approach.tex]: Indeed. Undone.

Comment (TheRumle) [inline report/src/introduction/agile_approach.tex]:
```suggestion
In incremental agile approaches, system development is considered a series of increments, each adding functionality to the system.
```

File: report/src/introduction/agile_approach.tex (hunk 1 above)
Comment (TheRumle) [reply report/src/introduction/agile_approach.tex:3]: I think that saying it allows for faster development is a bit of a stretch though. When I read that I think 'okay, so why is this not something everyone does?'
Maybe by saying 'these methods allows for fast adaptation to rapidly changing user needs and therefore faster development of software products where this is the case.'?

File: report/src/introduction/agile_approach.tex [hunk 11]
```diff
@@ -1,29 +1,27 @@
...
-\subsection*{Scrum Framework}
-Agile is a set of values, and does not present practical activities which a process model would consist of\cite{sutherlandScrumArtDoing2014}.
-Therefore, the team has decided on Scrum\cite{scrumdotorg}, which is an agile framework that puts the values to practice.
-This decision was made due to prior experience with using the framework amongst team members.
+\subsection{Scrum Framework}
+Agile is a set of values and does not present practical activities which a process model would consist of\cite{sutherlandScrumArtDoing2014}.
+Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum is chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}
 
-Scrum introduces a variety of new practices and roles, which we will describe in the following sections.
+Scrum introduces a number of practices and roles, which is described in the following sections.
```
Comment (TheRumle) [inline report/src/introduction/agile_approach.tex:14]:
```suggestion
Scrum introduces a number of practices and roles, which are described in the following sections. 
```

File: report/src/introduction/agile_approach.tex (hunk 9 above)
Comment (TheRumle) [reply report/src/introduction/agile_approach.tex:61]:
```suggestion
\subsubsection*{Scrum Process}
In summary, Scrum involves repeatedly taking backlog items and working on them during a sprint.
```

File: report/src/problem_analysis/understanding.tex [hunk 12]
```diff
@@ -0,0 +1,31 @@
+To establish knowledge about the domain and issues with the current system for managing meeting arrangements, two interviews with the stakeholders was conducted.  
+The goals of these interviews were twofold: 
+First and foremost, we wanted  to understand the current issues the stakeholders are facing. 
+To figure this out we conducted an unstructured interview\cite{robson2002real}.
```
Comment (chhoumann) [inline report/src/problem_analysis/understanding.tex:4]: Should we say why we chose an unstructured interview?

i.e. to optimize for unexpected discoveries

File: report/src/problem_analysis/understanding.tex [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+Secondly, we wanted to understand what the stakeholder sees as minimum requirements for a potential solution to their problem through an interview.
+This interview was based on ideas provided to us by the stakeholder during the initial contact.
+Due to not knowing the specifics about the problem, we could not create close-ended questions for a structured interview, and thus chose to use a semi structured interview focusing on open-ended questions. This allowed us to ask complementary questions to clarify answers \cite{InterviewsNHS}.
+
+
+\section{Understanding the problem} % What constitutes a solution for 
+During the initial interview, the stakeholders provided us with insight into what their current needs are, and why their current system does not satisfy these needs.
+First and foremost, the stakeholder have issues with arranging meetings throughout their day to day life.
```
Comment (chhoumann) [inline report/src/problem_analysis/understanding.tex:12]:
```suggestion
First and foremost, the stakeholder have issues with arranging meetings throughout their day.
```

File: report/src/problem_analysis/understanding.tex [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+When employees of Mariendal IT schedules meetings they sometimes experience problems with not having or not being able to locate available rooms.
```
Comment (chhoumann) [inline report/src/problem_analysis/understanding.tex:13]:
```suggestion
When employees of Mariendal IT schedules meetings, they sometimes experience problems with not having or not being able to locate available rooms.
```
